fetch/range/resources/long-wav.py

```diff
@@ -118,12 +118,11 @@ def main(request, response):
     bytes_remaining_to_send -= len(initial_write)
 
     while bytes_remaining_to_send > 0:
-        if not response.writer.flush():
-            break
-
         to_send = b'\x00' * min(bytes_remaining_to_send, sample_rate)
         bytes_remaining_to_send -= len(to_send)
 
-        response.writer.write(to_send)
+        if not response.writer.write(to_send):
+            break
```

Comment:
Note that this slightly changes the semantics (in the edge case when the first `write` outside of the `while` loop fails), but it should be fine I think (if the first `write` fails then we'd now call a second `write` that'd fail immediately). Just in case, have you verified the test resutls?

Reply:
Yes, I verified the test results locally.